#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_gateway.dpatch by Walter, dg9ep @ vdrportal.de
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes problems when ending a telnet session with ctrl-c
## DP: see: http://www.vdr-portal.de/board/thread.php?threadid=49680

@DPATCH@
diff -urNad vdr-plugin-control-0.0.2a~/gateway.c vdr-plugin-control-0.0.2a/gateway.c
--- vdr-plugin-control-0.0.2a~/gateway.c	2004-07-02 21:24:22.000000000 +0200
+++ vdr-plugin-control-0.0.2a/gateway.c	2007-04-12 22:51:16.000000000 +0200
@@ -36,7 +36,7 @@
   _pTelnet	(0),
   _pFormatter   (0)
 {
-  gl_CtrlGatewayCol.Add(this);
+//  gl_CtrlGatewayCol.Add(this);
 
 //  if (! cKbdRemoteForControl::_pRemote)
 //    cKbdRemoteForControl::_pRemote = new cKbdRemoteForControl();
@@ -163,8 +163,8 @@
 
 //  UnprepareSocket();
 
-  if (! _terminating)
-    gl_CtrlGatewayCol.Del(this);
+//  if (! _terminating)
+//    gl_CtrlGatewayCol.Del(this);
 
   conDSYSLOG("gateway thread ended (pid=%d)", getpid());
 conLOG_GATEW("gateway thread ended (pid=%d)", getpid());
